### PR TITLE
Add base api framework

### PR DESCRIPTION
**Pull request by JorgeRCV**

### CODE REVIEW

**Review by SaulCM — APPROVED** [review]

**Review by GustavoZHA — APPROVED** [review]

**Review by Raymundogs — APPROVED** [review]

**Review by cristianfchq — APPROVED** [review]

File: src/main/java/core/api/ApiRequestSpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import core.utils.Supplier;
+
+public class ApiRequestSpecificationProvider {
+
+    /**
+     * Checks if an ApiRequestField is not null and adds the specification if so.
+     *
+     * @param addSpecification
+     * @param getRequestField
```
Comment (GustavoZHA) [inline src/main/java/core/api/ApiRequestSpecificationProvider.java:24]: Add @autors tag

File: src/main/java/core/utils/Supplier.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
```
Comment (GustavoZHA) [inline src/main/java/core/utils/Supplier.java:9]: Add authors

File: src/main/java/core/api/ApiRequest.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+     * @return the ContentType Enum
+     */
+    public ContentType getContentType() {
+        return contentType;
+    }
+
+    /**
+     * Sets the ContentType
+     *
+     * @param contentType the content type Enum
```
Comment (cristianfchq) [inline src/main/java/core/api/ApiRequest.java:266]: Add @autors tag

File: src/main/java/core/api/ApiRequest.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+     */
+    public Map<String, String> getParams() {
+        return params;
+    }
+
+    /**
+     * Gets the ContentType
+     *
+     * @return the ContentType Enum
+     */
```
Comment (cristianfchq) [inline src/main/java/core/api/ApiRequest.java:258]: Add @autors tag

File: src/main/java/core/api/ApiRequestSpecificationProvider.java (hunk 1 above)
Comment (SaulCM) [reply src/main/java/core/api/ApiRequestSpecificationProvider.java:24]: Done.

File: src/main/java/core/utils/Supplier.java (hunk 2 above)
Comment (SaulCM) [reply src/main/java/core/utils/Supplier.java:9]: Done.

File: src/main/java/core/api/ApiRequest.java (hunk 3 above)
Comment (SaulCM) [reply src/main/java/core/api/ApiRequest.java:266]: Done.

File: src/main/java/core/api/ApiRequest.java (hunk 4 above)
Comment (SaulCM) [reply src/main/java/core/api/ApiRequest.java:258]: Done.

**Review by Raymundogs — APPROVED** [review]

**Pull request merged**